packages/beacon-node/src/chain/produceBlock/produceBlockBody.ts

```diff
@@ -223,6 +229,23 @@ export async function produceBlockBody<T extends BlockType>(
         fetchedTime,
       });
 
+      const targetGasLimit = this.executionBuilder.getValidatorRegistration(proposerPubKey)?.gasLimit;
+      if (!targetGasLimit) {
+        // This should only happen if cache was cleared due to restart of beacon node
+        this.logger.warn("Failed to get validator registration, could not check header gas limit", {
+          slot: blockSlot,
+          proposerIndex,
+          proposerPubKey: toPubkeyHex(proposerPubKey),
+        });
+      } else {
+        const parentGasLimit = (currentState as CachedBeaconStateBellatrix).latestExecutionPayloadHeader.gasLimit;
+        const expectedGasLimit = getExpectedGasLimit(parentGasLimit, targetGasLimit);
+
+        if (builderRes.header.gasLimit !== expectedGasLimit) {
```

Comment:
i think many builders/relays might  have bugs over here initially.

we should relax it and make sure its between [parent limit, exected limit] or [expected limit, parent limit] interval and then warn if its not equal to expected

Reply:
> i think many builders/relays might have bugs over here initially.

Running this branch on Holesky right now but it's hard to validate if gas target is not moving

> we should relax it and make sure its between [parent limit, exected limit] or [expected limit, parent limit]

Right so in this case we should definitely reject as the block would be invalid, we can relax for now and observe this for a while on mainnet, eventually we can enforce the strict check